pkg/healthcheck/healthcheck.go

```diff
@@ -1020,7 +1027,16 @@ func (hc *HealthChecker) checkClusterRoles(shouldExist bool) error {
 	}
 
 	objects := []runtime.Object{}
+
 	for _, item := range crList.Items {
+
+		if hc.Options.NoInitContainer {
+			label := item.ObjectMeta.Labels[linkerdCniResourceLabel]
+			if label == "true" {
+				continue
```

Comment:
TIOLI: Given this is repeated in the functions below, I'd DRY it into a separate function...